extras/check_code_style.sh

```diff
@@ -0,0 +1,228 @@
+
+Options:
+
+    --help               Print program options and exit.
+    --incremental        Do analysis one file after another.
+    --file=/path/to/file Perform the static analysis on this file only.
+    --git-start=SHA      Enter the SHA from which git starts the diff.
+                         (default=master)
+    --git-end=SHA        Enter the SHA from which git ends the diff.
+                         (default=HEAD)
+    --cppcheck=exe       Enter the executable that is used for cppcheck. Due to
+                         a bug in previous versions we require at least version
+                         1.69, which is also used in our TravisCI setup.
+                         (default=cppcheck)
+    --clang-format=exe   Enter the executable that is used for clang-format.
+                         Due to formatting differences in version 3.6 and 3.7
+                         we require version 3.6, which is also used in our
+                         TravisCI setup. (default=clang-format-3.6)
+    --vera++=exe         Enter the executable that is used for vera++.
+                         (default=vera++)
+EOF
+
+    exit $1
+}
+
+#
+# bail_out message
+#
+bail_out ()
+{
+    echo "$1"
+    exit 1
+}
+
+while test $# -gt 0 ; do
+    case "$1" in
+        --help)
+            usage 0
+            ;;
+        --incremental)
+            INCREMENTAL=true
+            ;;
+        --file=*)
+            ONLY_FILE="$( echo "$1" | sed 's/^--file=//' )"
+            if [ ! -f $ONLY_FILE ]; then
+              bail_out "This is not a valid file: $ONLY_FILE"
+            fi
+            ;;
+        --git-start=*)
+            START_SHA="$( echo "$1" | sed 's/^--git-start=//' )"
+            ;;
+        --git-end=*)
+            END_SHA="$( echo "$1" | sed 's/^--git-end=//' )"
+            ;;
+        --cppcheck=*)
+            CPPCHECK="$( echo "$1" | sed 's/^--cppcheck=//' )"
+            ;;
+        --vera++=*)
+            VERA="$( echo "$1" | sed 's/^--vera++=//' )"
+            ;;
+        --clang-format=*)
+            CLANG_FORMAT="$( echo "$1" | sed 's/^--clang-format=//' )"
+            ;;
+        *)
+            usage 1 "$1"
+            ;;
+    esac
+    shift
+done
+
+#
+# sed has different syntax for extended regular expressions
+# on different operating systems:
+# BSD: -E
+# other: -r
+#
+EXTENDED_REGEX_PARAM=r
+/bin/sh -c "echo 'hello' | sed -${EXTENDED_REGEX_PARAM} 's/[aeou]/_/g' "  >/dev/null 2>&1 || EXTENDED_REGEX_PARAM=E
+
+echo "Executing check_code_style.sh"
+echo "Tooling:"
+
+# check vera++ is set up appropriately
+$VERA ./nest/main.cpp >/dev/null 2>&1 || usage 1 "Executable $VERA for vera++ is not working!"
+$VERA --profile nest ./nest/main.cpp >/dev/null 2>&1 ||  bail_out "No profile called nest for vera++ installed. See https://nest.github.io/nest-simulator/coding_guidelines_c++#vera-profile-nest"
+echo "vera++ version: `vera++ --version`"
+
+# check cppcheck 1.69 is installed correctly
```

Comment:
I think it would be useful to comment in the code why we require 1.69 (and clang 3.6) exactly.